wandb/sdk/artifacts/artifact.py

```diff
@@ -2436,7 +2436,7 @@ def link(
             return api._artifact(linked_path)
         except Exception as e:
             wandb.termerror(f"Error fetching link artifact after linking: {e}")
```

Comment:
We shouldn't print anything if we're letting the error bubble. You can just remove this try-except.